workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -668,4 +642,73 @@ export class DynamicAuthorizationService {
     } while (paginationToken);
     return identityPermissions;
   }
+
+  private async _getAllIdentityPermissionsByIdentity(params: {
+    identityId: string;
+    identityType: IdentityType;
+  }): Promise<IdentityPermission[]> {
+    const { identityId, identityType } = params;
+    let paginationToken = undefined;
+    let identityPermissions: IdentityPermission[] = [];
+    //paginate through response
+    do {
+      // get all identity permissions associated to an identity
+      const response: GetIdentityPermissionsBySubjectResponse = await this.getIdentityPermissionsByIdentity({
+        paginationToken,
+        identityId,
+        identityType
+      });
+      paginationToken = response.paginationToken;
+      identityPermissions = identityPermissions.concat(response.data.identityPermissions);
+    } while (paginationToken);
+    return identityPermissions;
+  }
+  /**
+   * Delete all identity permissions associated to an identity
+   * @param params - requires an IdentityId, IdentityType, and AuthenticatedUser
+   *
+   * @throws - {@link RetryError} if an error is encountered during deletion
+   */
+  private async _deleteAllIdentityPermissionsByIdentity(params: {
+    identityId: string;
+    identityType: IdentityType;
+    authenticatedUser: AuthenticatedUser;
+  }): Promise<void> {
+    const { identityId, identityType, authenticatedUser } = params;
+    try {
+      const identityPermissions = await this._getAllIdentityPermissionsByIdentity({
+        identityId,
+        identityType
+      });
+      const deleteIdentityPermissionsPromises = _.chunk(identityPermissions, 100).map(
+        (chunkedIdentityPermissions) => {
+          return this.deleteIdentityPermissions({
+            authenticatedUser,
+            identityPermissions: chunkedIdentityPermissions
+          });
+        }
+      );
+      await Promise.all(deleteIdentityPermissionsPromises);
+    } catch (error) {
+      throw new RetryError(error.message);
+    }
+  }
+  /**
+   * Check to see if a group has a status
+   * @param params - groupId required to check
+   */
+  private async _hasGroupStatus(params: { groupId: string }): Promise<boolean> {
```

Comment:
Do we need this function now that we're throwing `GroupNotFoundError` in the `getGroupStatus` function?

Reply:
calling getGroupStatus without using the response and waiting to see if it throws feels rather not intuitive. I think keeping this will be clearer and cleaner